R/convertMT.R

```diff
@@ -0,0 +1,158 @@
+#' @title Convert migration times to effective mobility
+#'
+#' @description
+#'
+#' `convertMT` uses a list of known EOF markers with to convert migration times (MTs)
+#'     to effective mobilities (µeff). 
+#'     By using the information of the EOF markers (which is their MT and µeff), 
+#'     it is possible to transform the MT-scale (for consistency called rtime) 
+#'     into the µeff scale. This is used in CE-MS to overcome MT shifts originating 
+#'     from EOF fluctuations in different runs.
+#'
+#' @param x `numeric` vector with migration times
+#'
+#' @param y `data.frame`data.frame containing three columns, where the first
+#'     holds the ID of the respective markers, the second the determined migration 
+#'     time (here referred to as `rtime`) of the EOF marker in the same run in which 
+#'     the migration time is going to be transformed and the third column the 
+#'     respective mobility of the EOF markers. Each row hold the values for one 
+#'     EOF marker. The minimum of required markers is one. 
+#'
+#' @param method `character` single character that defines if the transformation 
+#' is either performed with a single ("single") EOF marker or multiple ("multiple").
+#'
+#' @param FUN `function` function defining how the conversion is performed,
+#'     either a single EOF marker is used (convertSingle) or multiple markers 
+#'     (convertMultiple)
+#'
+#' @return `numeric` vector of same length as x with effective mobility values. 
+#'
+#' @author Liesa Salzer
+#'
+#' @export
+#'
+#' @examples
+#' 
+#' 
+#' rtime <- c(10,20,30,40,50,60,70,80,90,100)
+#' marker <- data.frame(markerID = c("marker1", "marker2"),
+#'   rtime = c(20,80),
+#'   mobility = c(0, 2000))
+#' 
+#' convertMT(rtime, marker, method = "multiple")
+#' convertMT(rtime, marker, method = "single", U = 30, L = 90, markerID = "marker1")
+
+convertMT <- function(x, y, 
+                      method = c("single", "multiple"), ...) {
+  # sanity checks
+  if(missing(x)) {
+    stop("Missing vector 'x' with migration times")}
+  if(missing(y)) {
+    stop("Missing data.frame 'y' with marker information")}
+  if(!ncol(y) == 3) {
+    stop("'y' requires exact three columns")}
+  if(!all(c("markerID","rtime","mobility") %in% colnames(y))) {
+    stop("Missing column 'markerID', 'rtime', 'mobility' or all")}
+  if(missing(method)) {
+    stop("'method' is missing")}
+  if(length(method) != 1) {
+    stop("Conversion method is either 'single' or 'multiple'")}
+  if(!method %in% c("single", "multiple")) {
+    stop("Conversion method is either 'single' or 'multiple'")}
+  if(!is.numeric(x)) {
+    stop("'x' needs to be numeric")}
+  if(!is.numeric((y$rtime))) {
+    stop("'rtime' entries in 'y' needs to be numeric")}
+  if(!is.numeric((y$mobility))) {
+    stop("'mobility' entries in 'y' needs to be numeric")}
+  
+  
+  
+  # Change time-scale from sec to min
+  if(any(x > 60)){ 
```

Comment:
Please add as space between the `if` and the opening parenthesis and before the opening `{`. Also put the closing `}` on its own new line:

```
if(any(x > 60)) { 
x <- x/60  
}
```

Reply:
There's `MsCoreUtils::formatRt()` that addresses some retention time conversions. If other conversions are needed, we should consider adding them to `MsCoreUtils`.